modules/social_features/social_comment/social_comment.module

```diff
@@ -392,3 +392,13 @@ function social_comment_preprocess_pager(&$variables) {
     $items['pages'][$key]['href'] = preg_replace($pattern, '', $item['href']);
   }
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_delete().
+ */
+function social_comment_comment_delete(CommentInterface $comment): void {
```

Comment:
We shouldn't fix it this way.

This should be happening by default in Drupal core. See: https://git.drupalcode.org/project/drupal/-/blob/9.4.x/core/modules/comment/src/Entity/Comment.php#L188

If it is not happening, which seems the case here, then we need to find the root cause. Check https://www.drupal.org/project/drupal/issues/3232001 if this helps.

Reply:
@navneet0693 Yes, it should but cache tag invalidation works only for the deleted entity. We show the count of comments on the node's displays so we need to invalidate cache tags from the commented entity, so I'll add this process to the `postDelete` method in the overrides comment entity [class](https://github.com/goalgorilla/open_social/blob/bugfix/3258809/modules/social_features/social_comment/src/Entity/Comment.php) instead of the hook.

Reply:
@dmytriikaiun I do think that this can be added in the Drupal core class itself. What do you think?